wcfsetup/install/files/lib/data/media/Media.class.php

```diff
@@ -310,4 +312,23 @@ public static function getThumbnailSizes()
     {
         return static::$thumbnailSizes;
     }
+
+    #[\Override]
+    public function getImageData(?int $minWidth = null, ?int $minHeight = null): ?ImageData
+    {
+        if ($minWidth !== null || $minHeight !== null) {
+            foreach (\array_keys(self::$thumbnailSizes) as $size) {
+                if ($minWidth !== null && $minWidth > $this->getThumbnailWidth($size)) {
+                    continue;
+                }
+                if ($minHeight !== null && $minHeight > $this->getThumbnailHeight($size)) {
+                    continue;
+                }
+
+                return new ImageData($this->getThumbnailLink($size), $this->getThumbnailWidth($size), $this->getThumbnailHeight($size));
+            }
+        }
+
+        return new ImageData($this->getLink(), $this->width, $this->height);
```

Comment:
This has the same issue as the attachment implementation: It will return an image even if it does not satisfy the width and/or height constraints.